components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java

```diff
@@ -216,10 +219,13 @@ public void storeAccessToken(String accessToken, String consumerKey,
             prepStmt.setString(11, accessTokenDO.getTokenState());
             prepStmt.setString(12, accessTokenDO.getTokenType());
             prepStmt.setString(13, accessTokenDO.getTokenId());
-            prepStmt.setString(14, persistenceProcessor.getProcessedClientId(consumerKey));
+            prepStmt.setString(14, accessTokenDO.getGrantType());
+            prepStmt.setString(15, persistenceProcessor.getProcessedClientId(consumerKey));
```

Comment:
TOKEN_ID is 14th parameter and GRANT_TYPE is 15th parameter. check with the sql query